tests/func/test_recognizers.py

```diff
@@ -18,6 +18,8 @@ def test_parse_list_of_integers():
     """
 
     def int_less_than_five(input, pos):
+        if len(input) <= pos:
+            return None
```

Comment:
I understand that recognizing stuff at the end of the input is necessary for GLR to investigate all possibilities but this would add a lot of boilerplate code as recognizers commonly use `input[pos]` at the beginning so each of them would need this guard.

I would consider calling recognizer at the end of the input only if the terminal is marked that it could match empty. E.g. 

```
myterminal: {empty};
```
Such terminals would match at the end of the input implicitly and the recognizer won't be called.
This would make stuff more backward compatible while still enabling user to leverage full GLR parsing with a bit of change in the grammar. Thoughts?

Reply:
You got the point here. I agree that probably the decorators are the way to go. Especially as we already have a [decorator for collecting recognizers](http://www.igordejanovic.net/parglare/stable/grammar_modularization/#grammar-file-recognizers). The factory for it could be extended with a param to produce the decorator that will implement the boilerplate code above. I'll see to take care of that.

For the second point, since we are still in beta phase (pre 1.x) we are fine with backwards incompatibilities as long as we state that clearly in the `HISTORY.md`.

I'll see to test this change some more in the next days. Due to my schedule, probably won't be able before the weekend. If all goes well, for this to be merged, we should need some docs update and probably a test that verifies turning implicit lexical disambiguation on/off.